tests/api/fuzzer.py

```diff
@@ -0,0 +1,319 @@
+import argparse
+import json
+import logging
+import random
+import re
+import requests
+import sys
+import posixpath
+import urllib.parse
+
+
+#######################################################################################################################
+### Utilities
+
+def some(s):
+    return random.choice(sorted(s))
```

Comment:
Casting to a list is non-deterministic and will result in a different order. If the `--reseed` option has not been passed the order should be repeatable, the sort is just a way of picking an arbitrary deterministic order.

Reply:
So? Why is a deterministic order needed here tho? Why should it be repeatable? The whole point of a fuzz test is non-determinism. A non-deterministic order is just `more_randomness` in the returned element, as far as I'm concerned and that's a good thing. Maybe I'm missing something tho.

Reply:
The final version of the fuzzer after #7165 is intended to operate in several modes (or depths). The smaller sets are designed to be fast validations of netdata rather than extensive explorations of the parameter space. For these smaller tests we can put them straight into the build pipeline and repeatability will mean that we can rely on them - i.e. changing some code and retriggering the tests should not reveal a different set of problems to the one that an engineer is currently working on. For the larger tests we want to get lucky, so maximizing the randomness seems appropriate. 

At this point the tool is not really a fuzzer -> they aim for a large state space to overcome the single bit of information per test ("did the SUT  crash when we fuzzed it?"). Now that the swagger validation is up and running this is actually something more similar to a model-checker in that we retrieve a large amount of information about the SUT and what it did during the test sequence, i.e. it didn't crash but could we parse the output from the set of API calls to check that they are all valid. For this part to be useful it seems that repeatability is more desirable.

After #7165 it will glue both of these modes into the stress-tester. I'm not entirely sure what kind of tool that is, but if we leave it running against a build for a few hours (e.g. have a bot pull the nightly and run it on a schedule) then it should produce a high degree of confidence that the nightly is behaving as we expect.

Reply:
> At this point the tool is not really a fuzzer 

This. ;)